src/services/dashboards/widgets/widget-helper.ts

```diff
@@ -107,3 +113,59 @@ export const getRefinedXYChartData = (
     });
     return sortBy(chartData, categoryKey);
 };
+
+/**
+ * @name getRefinedDataTableFields
+ * @description Get refined PDataTable fields.
+ */
+export const getRefinedDataTableFields = (granularity: Granularity, dateRange: DateRange): DataTableFieldType[] => {
+    if (!granularity || !dateRange?.end) return [];
+    const dateFields: DataTableFieldType[] = [];
+    const start = dayjs.utc(dateRange.start);
+    const end = dayjs.utc(dateRange.end);
+
+    const timeUnit = getTimeUnitByPeriod(granularity, dayjs.utc(dateRange.start), dayjs.utc(dateRange.end));
```

Comment:
```suggestion
    const timeUnit = getTimeUnitByPeriod(granularity, start, end);
```
It looks like codes are duplicated here ..

Reply:
oh Thanks!!!!!